src/hooks/useShowTrailer.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+
+export default function useShowTrailer({
+  seriesTrailerUrl,
+  movieTrailerUrl,
+  isChecked,
+}) {
+  const [showTrailer, setShowTrailer] = useState(false);
+
+  useEffect(() => {
+    if (movieTrailerUrl?.length !== 0) {
+      setShowTrailer(true);
+    }
+    if (movieTrailerUrl?.length === 0) {
+      setShowTrailer(false);
+    }
+    if (seriesTrailerUrl?.length !== 0) {
+      setShowTrailer(true);
+    }
+    if (seriesTrailerUrl?.length === 0) {
+      setShowTrailer(false);
+    }
+    if (isChecked === true && movieTrailerUrl?.length === 0) {
+      setShowTrailer(false);
+    }
+    if (isChecked === true && seriesTrailerUrl?.length === 0) {
+      setShowTrailer(false);
+    }
+    if (isChecked === false) {
+      setShowTrailer(false);
+    }
+  }, [isChecked, movieTrailerUrl?.length, seriesTrailerUrl?.length]);
+
+  return { showTrailer };
+}
```

Comment:
Hi, so @mbosselmann and I had a look and we think that you could make your life a bit easier by making this hook even more custom. What you want is to check whether an URL is there or not. I don't think the hook needs to know about `movieTrailerUrl` and `seriesTrailerUrl`. It could simply know about a `trailerUrl`, then you would save a few `if` statements. If you have any questions about that, please get back to us 🙂 Also, we didn't test our theory 😅